experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py

```diff
@@ -3237,6 +3089,435 @@ def test_validate_feature_versioned_localized(self):
+        self,
+        min_version,
+        max_version,
+        expected_error,
+    ):
+        application = NimbusExperiment.Application.FENIX
+
+        feature = NimbusFeatureConfigFactory.create(
+            application=application,
+            slug="FEATURE",
+            name="FEATURE",
+            schemas=[
+                NimbusVersionedSchemaFactory.build(version=None, schema=None),
+                NimbusVersionedSchemaFactory.build(
+                    version=self.versions[(120, 0, 0)], schema=None
+                ),
+                NimbusVersionedSchemaFactory.build(
+                    version=self.versions[(122, 0, 0)], schema=None
+                ),
+            ],
+        )
+
+        NimbusFeatureConfigFactory.create(
+            application=application,
+            schemas=[
+                NimbusVersionedSchemaFactory.build(version=None, schema=None),
+                NimbusVersionedSchemaFactory.build(
+                    version=self.versions[(120, 0, 0)], schema=None
+                ),
+                NimbusVersionedSchemaFactory.build(
+                    version=self.versions[(121, 0, 0)], schema=None
+                ),
+                NimbusVersionedSchemaFactory.build(
+                    version=self.versions[(122, 0, 0)], schema=None
+                ),
+            ],
+        )
+
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.CREATED,
+            targeting_config_slug=NimbusExperiment.TargetingConfig.NO_TARGETING,
+            application=application,
+            firefox_min_version=min_version,
+            firefox_max_version=max_version,
+            feature_configs=[feature],
+        )
+
+        for branch in experiment.treatment_branches:
+            branch.delete()
+
+        serializer = NimbusReviewSerializer(
+            experiment,
+            data=NimbusReviewSerializer(experiment, context={"user": self.user}).data,
+            context={"user", self.user},
+        )
+
+        self.assertFalse(serializer.is_valid())
+        self.assertTrue(
+            serializer.errors["reference_branch"]["feature_values"][0]["value"][
+                0
+            ].startswith(expected_error),
+            serializer.errors,
+        )
+
+    @patch(
+        "experimenter.features.manifests.nimbus_fml_loader.NimbusFmlLoader.get_fml_errors",
+    )
+    def test_fml_validate_feature_versioned_truncated_range(
+        self,
+        mock_fml_errors,
+    ):
+        mock_fml_errors.return_value = [
+            NimbusFmlErrorDataClass(
+                line=1,
+                col=0,
+                message="Incorrect value!",
+                highlight="enabled",
+            ),
+        ]
+        application = NimbusExperiment.Application.IOS
+        min_version = NimbusExperiment.Version.FIREFOX_110
+        max_version = NimbusExperiment.Version.NO_VERSION
+
+        schema = json.dumps(
```

Comment:
This is not a JSON schema. JSON Schema is a specific data format for describing JSON objects. There are a few examples in this file you can look at:

- https://github.com/mozilla/experimenter/blob/71e34d01816190edf14a422cd630faccdf1e4a57/experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py#L31
- https://github.com/mozilla/experimenter/blob/71e34d01816190edf14a422cd630faccdf1e4a57/experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py#L46
- https://github.com/mozilla/experimenter/blob/71e34d01816190edf14a422cd630faccdf1e4a57/experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py#L2934

You probably want to use the last one, which describes an object that has a single field, "enabled", which is a boolean, and it does not accept any other undefined fields, which is the data type youre using on line 3239.

Lots of docs on [the JSON Schema website](https://json-schema.org/understanding-json-schema).